src/components/error_boundary/ErrorBoundary.tsx

```diff
@@ -0,0 +1,71 @@
+import React, { Component } from 'react';
+import type { ReactNode } from 'react';
+
+interface Props {
+  children: ReactNode;
+}
+
+interface State {
+  error?: Error;
+  hasError: boolean;
+}
+
+class ErrorBoundary extends Component<Props, State> {
+  constructor(props: Props) {
+    super(props);
+    this.state = {
+      error: undefined,
+      hasError: false,
+    };
+  }
```

Comment:
here you can do without a constructor at all, it will be much shorter
```suggestion
  state: State = { hasError: false, error: undefined };
```

Additionally, it is better to define a constant `const initState = { hasError: false, error: undefined }`